src/core/api.proto

```diff
@@ -30,6 +30,47 @@ package nvidia.inferenceserver;
 
 //@@.. cpp:namespace:: nvidia::inferenceserver
 
+//@@      .. cpp:var:: message InferSharedMemoryElement
+//@@
+//@@         The meta-data for the shared memory from which to read the input
+//@@         data and/or write the output data.
+//@@
+message InferSharedMemoryElement
+{
+  //@@    .. cpp:var:: string shm_key
+  //@@
+  //@@    The name of the shared memory region that holds the input data
+  //@@    (or where the output data should be written).
+  //@@
+  string shm_key = 1;
+
+  //@@    .. cpp:var:: uint64 offset
+  //@@
+  //@@       The offset from the start of the shared memory region.
+  //@@       element0 start = offset0, end = offset0 + size;
+  //@@
+  uint64 offset = 2;
+
+  //@@    .. cpp:var:: uint64 byte_size
+  //@@
+  //@@       Size of the element in the batch of the input tensor, in bytes.
+  //@@
+  uint64 byte_size = 3;
+}
+
+//@@    .. cpp:var:: message InferSharedMemory
+//@@
+//@@       Meta-data for I/O of batch from shared memory.
+//@@
+message InferSharedMemory
+{
+  //@@      .. cpp:var:: InferSharedMemoryElement element (repeated)
+  //@@
+  //@@         No. of elements required is the size of the batch.
```

Comment:
"One or more contiguous shared memory blocks that compose the entire shared memory"

How about change InferSharedMemoryElement to InferSharedMemoryBlock since that more accurately captures what it is.

Reply:
Yes I think Block would capture the meaning better